app/views/appointments/index.html.erb

```diff
@@ -1,72 +1,100 @@
-            </div>
-            <div class="row mt-2">
-              <div class="col-md-6 col-lg-4 overdue-button">
-                <% if phone_number = appointment.patient.phone_numbers.first&.number %>
-
-                    <a href="tel:<%= phone_number %>" class="btn btn-info btn-phone w-100">&#128222; <%= phone_number %></a>
+<div class="row">
+  <div class="col-md-6">
+    <%= bootstrap_form_with(url: appointments_path, method: :get, layout: :horizontal, class: "mt-4") do |form| %>
+      <div class="form-row">
+        <div class="form-group col-9">
+          <%= form.select :facility_id,
+                          current_admin.facilities.sort_by(&:name).map { |facility| [facility.name, facility.id] },
+                          {
+                            hide_label: true,
+                            include_blank: "All facilities",
+                            selected: @facility_id,
+                            wrapper: false
+                          },
+                          onchange: "this.form.submit();"
+          %>
+        </div>
+        <div class="form-group col-3">
+          <%= form.select :per_page,
+                          [20, 50, "All"],
+                          {
+                            help: "Per page",
+                            hide_label: true,
+                            selected: @per_page,
+                          },
+                          onchange: "this.form.submit();"
+          %>
+        </div>
+      </div>
+    <% end %>
+  </div>
+</div>
 
-                <% else %>
-                  <a href="3" class="btn btn-info disabled w-100">&lt; No mobile &gt;</a>
-                <% end %>
-              </div>
-              <div class="col-md-6 col-lg-4 overdue-form">
-                <%= bootstrap_form_with(model: appointment) do |form| %>
-                  <%= form.select :call_result,
-                                  {
-                                      "Patient responded" => [["Agreed to visit", "agreed_to_visit"]],
-                                      "No response" => [["Remind to call later", "remind_to_call_later"]],
-                                      "Remove from overdue list" => Appointment.cancel_reasons.map { |key, value| [key.humanize, value] }
-                                  },
-                                  {
-                                      prompt: "Result of follow-up...",
-                                      hide_label: true
-                                  },
-                                  onchange: "this.form.submit();"
-                  %>
-                <% end %>
-              </div>
-            </div>
+<div id="overdue-patients">
+  <%
+    appointments_with_patients = @appointments.select { |a| a.patient.present? }
+    if appointments_with_patients.present?
+  %>
+    <% appointments_with_patients.each do |appointment| %>
+      <section class="overdue-patient">
```

Comment:
We can also extract the overdue patient component to a partial.

Reply:
done